src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostConfiguration.cs

```diff
@@ -0,0 +1,190 @@
+        /// </summary>
+        public IList<IConfigurationSource> Sources => _sources;
+
+        /// <summary>
+        /// Returns the providers user to obtain configuration values.
+        /// </summary>
+        public IEnumerable<IConfigurationProvider> Providers => _providers;
+
+        /// <summary>
+        /// Gets a key/value collection that can be used to share data between the <see cref="IConfigurationBuilder"/>
+        /// and the registered <see cref="IConfigurationProvider"/>s.
+        ///
+        /// In this implementation, this largely exists as a way to satisfy the
+        /// requirements of the IConfigurationBuilder and is not populated by
+        /// <see cref="WebAssemblyHostConfiguration"/> with any meaningful info.
+        /// </summary>
+        public IDictionary<string, object> Properties { get; } = new Dictionary<string, object>();
+
+        /// <summary>
+        /// Gets or sets a configuration value in all providers that are currently
+        /// registered on the <see cref="WebAssemblyHostConfiguration"/> object.
+        /// </summary>
+        /// <param name="key">The configuration key.</param>
+        /// <returns>The configuration value.</returns>
+        public string this[string key] {
+            get {
+                for (var i = _providers.Count - 1; i >= 0; i--)
+                {
+                    var provider = _providers[i];
+
+                    if (provider.TryGet(key, out var value))
+                    {
+                        return value;
+                    }
+                }
+
+                return null;
+            }
+            set {
+                if (_providers.Count == 0)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                foreach (var provider in _providers)
+                {
+                    provider.Set(key, value);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
+        /// </summary>
+        /// <param name="key">The key of the configuration section.</param>
+        /// <returns>The <see cref="IConfigurationSection"/>.</returns>
+        /// <remarks>
+        ///     This method will never return <c>null</c>. If no matching sub-section is found with the specified key,
+        ///     an empty <see cref="IConfigurationSection"/> will be returned.
+        /// </remarks>
+        public IConfigurationSection GetSection(string key) => new ConfigurationSection(this, key);
+
+        public IEnumerable<IConfigurationSection> GetChildren() => GetChildrenImplementation(null);
+
+        /// <summary>
+        /// Gets the immediate descendant configuration sub-sections.
+        /// </summary>
+        /// <returns>The configuration sub-sections.</returns>
+        private IEnumerable<IConfigurationSection> GetChildrenImplementation(string path)
```

Comment:
`GetChildrenImplementation` isn't re-used and `path` is always null Any reason it couldn't be merged in to `GetChildren`?

Reply:
Done.